### ISSUE TITLE
Compatibility with top-level immutable store

### ISSUE THREAD

**Issue by neverfox** [tommikaikkonen/redux-orm#82]
redux-orm appears to require a mutable store, so doesn't work with total store immutability that comes from `redux-immutable` or `redux-seamless-immutable`.  Would it be possible to allow for a configuration that makes this possible (perhaps through user defined serialization/deserialization functions)?

**Comment by tommikaikkonen** [comment]
Redux ORM does not mutate state data. Beneath the session abstraction, a pointer is held on the most up to date state after each update, which makes it seem as if your mutating data. Mutations are only applied to objects that were created during the same session to improve efficiency.

In your reducer, you start a session from the state input argument, apply updates within the session abstraction, get the reference to the updated state, and return that. From the perspective of reducer caller, it's totally immutable.

Or maybe I'm missing something here?

**Comment by neverfox** [comment]
Yes, sorry. I'm talking about using ORM with something like Immutable.js or seamless-immutable. What I'm saying is that my whole redux state is frozen with seamless-immutable all the way that the top. When I tried to perform an action like `create` with ORM, I got errors about trying to mutate my immutable app state. When I reverted to vanilla JS state, all was well. So I'm trying to figure out what, if anything, I can do to make it work under those circumstances, or if ORM inherently requires that I not "freeze" my top-level state object.

**Comment by neverfox** [comment]
I should point out that I'm not using the new `ormReducer` pattern, but the static reducer method pattern (because I like to split up my reducers by model rather than have one uber reducer). So I'm just performing ORM actions without getting hold of the state anywhere.

**Comment by 23tux** [comment]
I just ran in the same issue. @neverfox How did you deal with it?

**Comment by TomBerriot** [comment]
Hi, @23tux  I have the error as you but without using Immutable libraries.

I'm running Redux ORM on React native. React Native has a fast refresh functionality which reloads very quickly on code changes https://facebook.github.io/react-native/docs/fast-refresh

Often when we change code other than JSX or style, we have this error appearing.
But I only got once the chance to see that the was coming from Redux ORM. Other times the error comes from native on js parsing.

![image](https://user-images.githubusercontent.com/11588007/68469768-19e48280-021b-11ea-920b-08a543396b22.png)

**Comment by Isaacmeedinaa** [comment]
@TomBerriot Hey Tom, were you able to fix this?
